### ISSUE TITLE
_ComputeLockFileReferences/RunResolvePackageDependencies should be faster when projects are up-to-date

### ISSUE THREAD

**Issue by davkean** [dotnet/sdk#1116]
Investigating up-to-date builds from ASP.NET, I've found that _ComputeLockFileReferences/RunResolvePackageDependencies is taking a non-trivial amount of time (> 17%) of the build time when a project is considered up-to-date.

```
109>        1 ms  CheckForImplicitPackageReferenceOverrides   1 calls
109>        1 ms  IncludeTransitiveProjectReferences         1 calls
109>        1 ms  ExpandSDKReferences                        1 calls
109>        1 ms  CoreCompile                                1 calls
109>        2 ms  GenerateBuildRuntimeConfigurationFiles     1 calls
109>        3 ms  _ComputeActiveTFMFileDependencies          1 calls
109>        3 ms  CopyFilesToOutputDirectory                 1 calls
109>        4 ms  ResolveLockFileReferences                  1 calls
109>        7 ms  _ComputeActiveTFMPackageDependencies       1 calls
109>       10 ms  _ComputeTFMOnlyFileDependencies            1 calls
109>       15 ms  GetTargetFrameworkProperties              79 calls
109>       19 ms  _GenerateCompileDependencyCache            1 calls
109>       20 ms  _CopyFilesMarkedCopyLocal                  1 calls
109>       25 ms  _ComputeTransitiveProjectReferences        1 calls
109>       27 ms  AssignProjectConfiguration                 1 calls
109>       27 ms  IncrementalClean                           1 calls
109>       32 ms  _SplitProjectReferencesByFileExistence     1 calls
109>       49 ms  GetCopyToOutputDirectoryItems              1 calls
109>      122 ms  _CleanGetCurrentAndPriorFileWrites         1 calls
109>      141 ms  ResolveProjectReferences                   1 calls
109>      368 ms  RunResolvePackageDependencies              1 calls <-- This bug
109>      912 ms  _ComputeLockFileReferences                 1 calls <-- This bug
109>     1630 ms  ResolveAssemblyReferences                  1 calls
109>     5970 ms  _GetProjectReferenceTargetFrameworkProperties  95 calls
```

We should investigate ways of caching/making this faster.

**Comment by jeromelaban** [comment]
This issue is very visible visual studio, where a non-modified project should not be touched. Large solutions get pretty long to build if everything is cross-targeted.

**Comment by mikeharder** [comment]
@nguerrera: Is this resolved by https://github.com/dotnet/sdk/pull/1857?

**Comment by nguerrera** [comment]
There is still more we can do, but #1857 should make this dramatically better.

**Comment by nguerrera** [comment]
I'm going to go ahead and close this as fixed by #1857. It will be best to reopen a more specific bug for any refinements beyond that should the new task/target still be a place to focus.

**Comment by nguerrera** [comment]
Actually, I'll reopen because design time build will still use RunResolvePackageDependencies and the slowness for the VS case and multi-targeting was mentioned.

**Comment by nguerrera** [comment]
A separate issue capturing the work for design-time case would be better. So I'm closing this as fixed by #1857 and #2020

**Comment by nguerrera** [comment]
DT part tracked by #2100
